tests/_test_msui/test_mscolab.py

```diff
@@ -901,6 +904,39 @@ def test_profile_dialog(self, qtbot):
             critbox.assert_called_once()
         assert not self.window.mscolab.profile_dialog.gravatarLabel.pixmap().isNull()
 
+    def test_upload_and_fetch_profile_image(self, qtbot):
+        self._connect_to_mscolab(qtbot)
+        self._login(qtbot, self.userdata[0], self.userdata[2])
+        self.window.mscolab.profile_action.trigger()
+
+        initial_pixmap = self.window.mscolab.profile_dialog.gravatarLabel.pixmap().toImage()
+
+        # Creating a new image
+        new_image = Image.new('RGB', (64, 64), color='cyan')
+        img_byte_arr = io.BytesIO()
+        new_image.save(img_byte_arr, format='JPEG')
+        img_byte_arr.seek(0)
+
+        # Path where the image will be temporarily stored
+        temp_image_path = os.path.join(os.getcwd(), 'new_profile_image.jpg')
+
+        # Saving the imaeg on path
+        with open(temp_image_path, 'wb') as f:
+            f.write(img_byte_arr.getvalue())
+
+        # Mocking the QFileDialog to select the image
+        with mock.patch('PyQt5.QtWidgets.QFileDialog.getOpenFileName', return_value=(temp_image_path, 'Image (*.jpg)')):
+            self.window.mscolab.upload_image()
+
+        qtbot.wait(1000)
+        self.window.mscolab.fetch_profile_image(refresh=True)
```

Comment:
I think there should be some assert for the updated pixmap before the `fetch_profile_image` call, because it is expected that `upload_image` updates the local images already, right?

This is related to not using `qtbot.wait`.

Reply:
Yes you are right, it would definitely make this test more robust